omrmakefiles/confighelpers.m4

```diff
@@ -67,7 +67,10 @@ AC_DEFUN([OMRCFG_CATEGORIZE_ARCH],
 AC_DEFUN([OMRCFG_CATEGORIZE_TOOLCHAIN],
 	[AC_LANG([C])
 	AS_IF([test "x$$1" = "x"],
-		AS_IF([test "$GCC" = "yes"], [$1=gcc]))
+		AS_IF([test "$GCC" = "yes"],
+			AS_IF(AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#ifndef __open_xl__
+#error not an openxl compiler
+#endif]])], [$1=openxl], [$1=gcc]))))
```

Comment:
Please explain this: Why should we test for OpenXL if `$GCC` is `yes`?

Reply:
I'm not sure exactly how the value of `$GCC` is determined, but it seems that with clang-based compilers (like OpenXL or xlclang) it ends up being set to "yes". So, before this change, when using OpenXL, `OMR_TOOLCHAIN` would end up being set to `gcc`. 

I initially put the OpenXL test in there to make sure that the difference between the `openxl` and `gcc` toolchains is caught, but I've now separated it out. This should still successfully detect when OpenXL is being used while being laid out in a more intuitive way.